pkg/server/http/http.go

```diff
@@ -39,6 +39,10 @@ func New(logger log.Logger, reg *prometheus.Registry, comp component.Component,
 	}
 
 	mux := http.NewServeMux()
```

Comment:
One small nit. We still allocate memory for a new mux even if we pass one. Is it better to use 
``` go
	var mux *http.ServeMux
	if options.mux != nil {
		mux = options.mux
	} else {
		mux = http.NewServeMux()
	}
```

Reply:
Right. As this is only called once in the lifecycle ever I'm personally fine with the current way this is implemented. Happy to update nonetheless if you think it's worth it.
The idea being, that the default Server works out of the box and then options can modify the defaults.

Reply:
Yea, that's not a big deal 👍🏽 Happy to merge.